stl/inc/format

```diff
@@ -1175,6 +1178,44 @@ public:
     }
 };
 
+template <class _CharT>
+constexpr size_t _Get_format_arg_type_storage_size(_Basic_format_arg_type _Type) {
+    switch (_Type) {
+    case _Basic_format_arg_type::_None:
+        throw format_error("_None type has no size.");
+    case _Basic_format_arg_type::_Int_type:
+        return sizeof(int);
+    case _Basic_format_arg_type::_UInt_type:
+        return sizeof(unsigned int);
+    case _Basic_format_arg_type::_Long_long_type:
+        return sizeof(long long);
+    case _Basic_format_arg_type::_ULong_long_type:
+        return sizeof(unsigned long long);
+    case _Basic_format_arg_type::_Bool_type:
+        return sizeof(bool);
+    case _Basic_format_arg_type::_Char_type:
+        return sizeof(_CharT);
+    case _Basic_format_arg_type::_Float_type:
+        return sizeof(float);
+    case _Basic_format_arg_type::_Double_type:
+        return sizeof(double);
+    case _Basic_format_arg_type::_Long_double_type:
+        return sizeof(long double);
+    case _Basic_format_arg_type::_Pointer_type:
+        return sizeof(void*);
+    case _Basic_format_arg_type::_CString_type:
+        return sizeof(const _CharT*);
+    case _Basic_format_arg_type::_String_type:
+        return sizeof(basic_string_view<_CharT>);
+    case _Basic_format_arg_type::_Custom_type:
+        return sizeof(typename basic_format_arg<basic_format_context<back_insert_iterator<basic_string<_CharT>>,
+                typename basic_string<_CharT>::value_type>>::handle);

```

Comment:
Isn't `typename basic_string<_CharT>::value_type` simply `_CharT`?

Does instantiating `basic_format_arg<basic_format_context<back_insert_iterator<basic_string<_CharT>>, typename basic_string<_CharT>::value_type>>` necessarily instantiate `basic_format_context` and `back_insert_iterator`? If so, it would be nice to avoid doing so unnecessarily. I think we'd be safe to simply assume that `sizeof(handle) == 2 * sizeof(void*)` given that users are forbidden to specialize nested types in general, and forbidden to specialize `basic_format_context` as well.

Reply:
yeah I thought about doing that. I think it should actually be `sizeof(void*) + sizeof(void (*)())` though

Reply:
> yeah I thought about doing that. I think it should actually be `sizeof(void*) + sizeof(void (*)())` though

On our platforms, `sizeof(void*)` is always the same as `sizeof(void(*)())`. That said, I don't mind the wordier and more precise formulation.